pkg/controller/controller_test.go

```diff
@@ -532,6 +532,7 @@ func createFakeNamedPVC(requestBytes int64, name string, userAnnotations map[str
 			Name:        name,
 			Namespace:   "fake-ns",
 			Annotations: annotations,
+			Finalizers:  []string{"volume.kubernetes.io/clone-protection"},
```

Comment:
The whole point of "clone protection finalizer" check is to ensure that the external-provisioner **adds** the finalizer when needed. Adding finalizer to the initial PVC will only make the tests pass, without actually testing anything.

A real fix will be far more complicated. You need to strudy when the controller adds the finalizer, when it removes the finalizer and how the unit test reads the finalizer. My guess would be that the unit test [reads](https://github.com/kubernetes-csi/external-provisioner/blob/1a7e9381439295969ad0336f1e21791f7dc3abe8/pkg/controller/controller_test.go#L6682-L6686) the PVC to check from an informer and the informer is stale. Try to read the PVC directly from the API server instead.